lib/methods/release-action.ts

```diff
@@ -0,0 +1,30 @@
+import { ReleaseActionProps, ReleaseActionTypes } from '../entities/release-action'
+import { PlainClientAPI } from '../plain/common-types'
+import { AsyncActionProcessingOptions, pollAsyncActionStatus } from './action'
+
+type PlainOptions = {
+  /** Used by the PlainClient to perform a poll for the BulkAction status */
+  plainClient: PlainClientAPI
+  spaceId: string
+  environmentId: string
+  releaseId: string
+  actionId: string
+}
+
+/** Waits for a ReleaseAction status to be either succeeded or failed.
+ * Used by the Plain client */
+export async function waitForReleaseActionProcessing<T extends ReleaseActionTypes = any>(
+  { plainClient, spaceId, environmentId, releaseId, actionId }: PlainOptions,
+  options?: AsyncActionProcessingOptions
+): Promise<ReleaseActionProps<T>> {
+  return pollAsyncActionStatus<ReleaseActionProps>(
+    async () =>
+      plainClient.releaseAction.get({
+        releaseId,
+        spaceId,
+        environmentId,
+        actionId,
+      }),
```

Comment:
**Question:** is `client.releaseAction.get({...})`, and `client.releaseAction.queryForRelease({...})` supposed to require `envId` and `spaceId` in the params?

Feels a bit confusing because it's not required in `environment.getReleaseAction`, and we could potentially get those from the client, no? 🤔 

<img width="500" alt="Screenshot 2021-05-20 at 07 25 14" src="https://user-images.githubusercontent.com/6163988/118924166-fdabc080-b93c-11eb-9741-01f8846ace0c.png">

Reply:
in `environment.getReleaseAction`, the environment + space is already part of the `environment` object, so no need to pass it

same for the plainClient, it's optional because we can set them in the plainClient initialisation (sounds confusing):


e.g.:

```
// plain  arguments: first accessToken, then "defaults" that you want the client to have
createClient(
  { accessToken: "123" }, { envId: 'master', spaceId: '222' }
)
```

Reply:
The line `getPlainClient(defaultParams)` is the one that sets it